slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,370 @@
+taxi_wflow <- workflow(tip ~ ., tree_spec)
+taxi_fit <- fit(taxi_wflow, taxi_train)
+```
+
+```{r more-setup}
+#| include: false
+library(countdown)
+
+options(width = 200)
+
+ggplot2::theme_set(ggplot2::theme_bw())
+```
+
+```{r}
+taxi_train
+```
+
+## Working with other models
+
+Some models can't handle non-numeric data (missing data)
+
+-   Linear Regression
+-   K Nearest Neighbors
+
+<br>
+
+::: fragment
+Some models struggle if numeric variables aren't scaled
+
+-   K Nearest Neighbors
+-   Anything using gradient descent
+:::
+
+## Types of needed preprocessing
+
+-   **dummy**: Do qualitative predictors require a numeric encoding?
+
+-   **zv**: Should columns with a single unique value be removed?
+
+-   **impute**: Does the model struggle with missing data?
+
+-   **decorrelate**: Does the model struggle with correlated predictors?
+
+-   **normalize**: Should predictors be centered and scaled?
+
+-   **transform**: Is it helpful to transform predictors to be more symmetric?
+
+::: footer
+<https://www.tmwr.org/pre-proc-table.html>
+:::
+
+## Two types of preprocessing
+
+![](images/fe_venn.svg){fig-align="center"}
+
+## Two types of preprocessing
+
+![](images/fe_venn_info.svg){fig-align="center"}
+
+## General definitions
+
+<br>
+
+* _Data preprocessing_ are the steps that you take to make your model **successful**
+* _Feature engineering_ are what you do to the original predictors to make the model do the **least work** to perform great 
+
+## Working with dates
+
+This data set originally contained a datetime variable. If given as a raw predictor, it is converted to an integer.
+
+It can be re-encoded as:
+
+* Days since a reference date
+* Day of the week
+* Month
+* Year
+* Leap year
+* Indicators for holidays
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute bottom="0" left="0" width="150" height="150"}
+
+*What other transformations could be do with the raw time variable?*
+
+<br>
+
+<br>
+
+Remember that the transformations are tied to the specific modeling problem
+
+```{r}
+#| echo: false
+countdown(minutes = 3, id = "datetime-split")
+```
+
+## two types of transformations
+
+::: columns
+::: {.column width="50%"}
+
+### static
+
+- sqrt, log, inverse
+- dummies with known levels
+- date time extractions
+
+:::
+
+::: {.column width="50%"}
+
+### trained
+
+- centering & scaling
+- Imputation
+- PCA
+- anything for unknown levels
+
+:::
+
+:::
+
+::: fragment
+Trained methods needs to calculate **sufficient information** to be applied again
+:::
+
+## recipes package
+
+extensible framework for pipeable sequences of feature engineering steps provides preprocessing tools to be applied to data
+
+::: {.incremental .highlight-last}
+- Modular + extensible
+- pipeable
+- Deferred evaluation
+- Isolates test data from training data
+- Can do things formulas can't
+:::
+
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
```

Comment:
I think the use of highlighting in the next few slides is really clever! Is there a way to make the code blocks look more like the regular R code blocks?

Reply:
I love this way of doing it! but it is super tedious and almost never worth it. I fully expect it to gain quarto support later

Reply:
Woah, yeah. This is awesome.